src/Microsoft.FeatureManagement.AspNetCore/DefaultHttpTargetingContextAccessor.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.AspNetCore.Http;
+using Microsoft.FeatureManagement.FeatureFilters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// Provides a default implementation of <see cref="ITargetingContextAccessor"/> that creates <see cref="TargetingContext"/> using info from the current HTTP request.
+    /// </summary>
+    public sealed class DefaultHttpTargetingContextAccessor : ITargetingContextAccessor
+    {
+        /// <summary>
+        /// The key used to store and retrieve the <see cref="TargetingContext"/> from the <see cref="HttpContext"/> items.
+        /// </summary>
+        public const string TargetingContextLookup = $"Microsoft.FeatureManagement.TargetingContext";
```

Comment:
It's debatable who should own this cache key. It's odd to see the more generic initializer and middleware reference the cache key from a very specific targeting context accessor.

Reply:
Agreed... I still am unable to come up with a satisfying place to put this. I suppose the owner could be the middleware- because the middleware's whole goal is to ensure that the `TargetingContext` is available at that key.

Reply:
The targeting context accessor doesn't need to share a key.

`private static object _cacheKey = new object();` should work fine.

Reply:
There's 3 classes that interact with HttpContext.Items for targeting context. `Initializer`, `Middleware`, and the `TargetingContextAccessor`.

`Initializer` depends upon the `Middleware`. The `Middleware` depends upon a `TargetingContextAccessor` existing. The `Middleware` and `Initializer` are tightly coupled- so they definitely need to share a key. The `TargetingContextAccessor` doesn't have to share the same key, but it reduces our HttpContext.Items footprint if it does- and is a clear shared cache location.

So for `Initializer` and `Middleware`, we are going to have a shared key- it could be either an object ref or the string key `"Microsoft.FeatureManagement.TargetingContext"`. I slightly prefer the string key since it's public (on the HttpContext.Items and as a public property). The key needs a home, and if it was just these two classes it would either be on the middleware or a shared constants class.

Now the `DefaultTargetingContextAccessor` comes in- which is our default implementation. It needs a cache key as well. It could either define it's own or used a shared constants class. 

I think the shared constants class makes things the most clear- as FM is defining where `TargetingContext` should live on `HttpContext.Items`. I ended up not liking adding the constants class because it introduces yet another public class, and I ended up going with the `DefaultTargetingContextAccessor`.